docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,153 @@
+
+[!code-csharp[StarterAutoMapper](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+That code maps properties of the `ISyndicationItem` object to the `NewsStoryViewModel` properties. You want the AutoMapper to provide the mapping using a constructor instead. Replace the above code with the following automapper configuration:
+
+[!code-csharp[FinishedViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+Notice that because this class is small, and you've examined carefully, you should turn on the `#nullable enable` directive above this class declaration. This change could have broken something, so it's worthwhile to run all the tests and test the application before moving on.
+
+The first set of changes showed you how to discover when the original design indicated that variables should not be set to null. The technique is referred to as **correct by construction**. You guarantee that an object and its properties cannot be null when it is constructed. The compiler's flow analysis provides assurance that those properties are not set to null after construction.
+
+Other times, the structure of a class provides different clues to the intent. Open the `Error.cshtml.cs` file in the `Pages` folder. The `ErrorViewModel` contains the following code:
+
+[!code-csharp[StarterErrorModel](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/Pages/Error.cshtml.cs#StartErrorModel)]
+
+Add the `#nullable enable` directive before the class declaration, and a `#nullable restore` directive after it. You'll get one warning that `RequestId` is not initialized. By looking at the class, you should decide that the `RequestId` property should be null in some cases. The existence of the `ShowRequestId` property indicates that missing values are possible. Because `null` is valid, add the `?` on the `string` type to indicate the `RequestId` property is a *nullable reference type*. The final class looks like the following example:
+
+[!code-csharp[FinishedErrorModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/Pages/Error.cshtml.cs#ErrorModel)]
+
+Check for the uses of the property, and you see that in the associated page, the property is checked for null before rendering it in markup. That's a safe use of a nullable reference type, so you're done with this class.
+
+## Fixing nulls causes change
+
+Sometimes, the fix for one set of warnings creates new warnings in related code. Let's see the warnings in action by fixing the `index.cshtml.cs` class Open the `index.cshtml.cs` file and examine the code. This file contains the code behind for the index page:
```

Comment:
"Sometimes" => "Almost always"? :)

Maybe not quite that strong, but I think it would be good to give the firm impression that if you feel you're playing whack-a-mole with warnings, that *doesn't* mean anything's wrong... unless you end up in a cycle.

Reply:
@jskeet "Frequently?"

Reply:
Works for me!